agent/engine/docker_task_engine_test.go

```diff
@@ -1146,3 +1146,27 @@ func TestEngineDisableConcurrentPull(t *testing.T) {
 	assert.False(t, dockerTaskEngine.enableConcurrentPull,
 		"Task engine should not be able to perform concurrent pulling for version < 1.11.1")
 }
+
+// TestTaskWithCircularDependency tests the task with containers of which the
```

Comment:
Indeed, this sort of cyclical dependency is being tested [here](https://github.com/aws/amazon-ecs-agent/blob/467c3d772d55e02ad618e4901afed1da1173496e/agent/engine/dependencygraph/graph_test.go#L88-L98).